engine/table/src/main/java/io/deephaven/engine/table/impl/locations/local/FileKeyValuePartitionLayout.java

```diff
@@ -80,9 +82,9 @@ public void findKeys(@NotNull final Consumer<TLK> locationKeyObserver) {
         try {
             Files.walkFileTree(tableRootDirectory.toPath(), EnumSet.of(FileVisitOption.FOLLOW_LINKS),
                     maxPartitioningLevels + 1, new SimpleFileVisitor<>() {
-                        final Set<String> takenNames = new HashSet<>();
-                        final List<String> partitionKeys = new ArrayList<>();
+                        final Set<String> partitionKeys = new LinkedHashSet<>(); // Preserve order of insertion
```

Comment:
I had to use a LinkedHashSet here to preserve the order of insertion of partition keys. Otherwise, I would need to maintain an additional list of partition keys along with takenNames on top of the new TIntObjectMap, which felt redundant. Let me know if you have suggestions.